snuba/querylog/__init__.py

```diff
@@ -28,24 +63,8 @@ def record_query(
         # circular dependency, where state would depend on the higher level
         # QueryMetadata class
         state.record_query(query_metadata.to_dict())
-
-        final = str(request.query.get_final())
-        referrer = request.referrer or "none"
-        timer.send_metrics_to(
-            metrics,
-            tags={
-                "status": query_metadata.status.value,
-                "referrer": referrer,
-                "parent_api": request.settings.get_parent_api(),
-                "final": final,
-            },
-            mark_tags={
-                "final": final,
-                "referrer": referrer,
-                "parent_api": request.settings.get_parent_api(),
-            },
-        )
-
+        _record_timer_metrics(request, timer, query_metadata)
+        _record_attribution_metrics(request, extra_data)
```

Comment:
How does extra_data get populated? There are expectations of it having specific fields but its type is `Mapping[str, Any]` without any explanation in the docstring

Reply:
It's actually a TypedDict elsewhere in the code (https://github.com/getsentry/snuba/pull/2342/files#diff-9523648ba19de16486059a3c7602cf46b28ed7065d2db716f81838e356d7d412R13)

I think it's not typed here to avoid dependencies.